nautobot/docs/development/apps/migration/ui-component-framework.md

```diff
@@ -0,0 +1,244 @@
+# Migrating to UI Component Framework
+A guide for app developers transitioning to the UI Component Framework.
```

Comment:
Let's make sure the documentation (possibly as separate file) also addresses best practices for *new* app development.

We probably need to make some updates to any number of existing app developer docs, probably starting with the files under `/docs/development/apps/api/views/`, to make sure the examples therein aren't outdated.

Reply:
I did some analysis of docs there and there are pretty-much accurate. I just have a couple of ideas for improvement:
1. Add direct information about preferred view class in `views/index.md` with link to the `nautobotuiviewset.md`
2. Add information to the `nautobot-generic-views.md` that this is outdated and link to the ui viewset.
3. Add information to the `nautobotuiviewset.md` # View Template Context that we have `panels` approach right now and link it.